core/src/Kokkos_CopyViews.hpp

```diff
@@ -1350,7 +1350,7 @@ contiguous_fill_or_memset(
     typename ViewTraits<DT, DP...>::const_value_type& value) {
 // On A64FX memset seems to do the wrong thing with regards to first touch
 // leading to the significant performance issues
-#ifndef KOKKOS_ARCH_A64FX
+#if !defined(KOKKOS_ARCH_A64FX) && !defined(KOKKOS_ENABLE_OPENMP)
   if (Impl::is_zero_byte(value))
     ZeroMemset<ExecutionSpace, View<DT, DP...>>(exec_space, dst, value);
   else
```

Comment:
I guess we only care about `Kokkos_ARCH_A64FX` for `OpenMP` and we still want to use `ZeroMemset` for other execution spaces (independent of the host architecture).
```suggestion
  if (Impl::is_zero_byte(value)
#ifdef KOKKOS_ENABLE_OPENMP
      && std::is_same_v<ExecutionSpace, Kokkos::OpenMP>
#endif
  )
    ZeroMemset<ExecutionSpace, View<DT, DP...>>(exec_space, dst, value);
  else
```